rasa/engine/validation.py

```diff
@@ -0,0 +1,324 @@
+import inspect
+import logging
+import typing
+from typing import Optional, Callable, Text, Tuple, Dict, Type, Any, Set
+
+import dataclasses
+import typing_utils
+
+from rasa.engine.exceptions import GraphSchemaValidationException
+from rasa.engine.graph import (
+    GraphSchema,
+    GraphComponent,
+    SchemaNode,
+    Fingerprintable,
+    ExecutionContext,
+)
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.nlu import components
+
+
+@dataclasses.dataclass
+class ParameterInfo:
+    """Stores metadata about a function parameter."""
+
+    type_annotation: Any
+    is_kwargs: bool
```

Comment:
Can we explain this with a comment or clearer name. It took me a while to realise this means `**kwargs` rather than any singular keyword argument.

Reply:
Could we be caught out if someone uses `*args`?

Reply:
> Can we explain this with a comment or clearer name. It took me a while to realise this means **kwargs rather than any singular keyword argument.

did both 👍🏻 

> Could we be caught out if someone uses *args?
 Write

Will add a test for this.

Reply:
I'm just ignoring variable length positional arguments now. Warning about them doesn't necessarily make sense as one might use them as a way to forbid any positional arguments (as described [here](https://stackoverflow.com/a/50565247))